cpp/posix_tcp.cpp

```diff
@@ -0,0 +1,107 @@
+
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <thread>
+#include <future>
+
+#include <fibre/protocol.hpp>
+
+
+#define TCP_RX_BUF_LEN	512
+
+class TCPBytesSender : public StreamSink {
+public:
+    TCPBytesSender(int socket_fd) :
+        _socket_fd(socket_fd)
+    {}
+
+    int process_bytes(const uint8_t* buffer, size_t length, size_t& processed_bytes) {
+        int bytes_sent = send(_socket_fd, buffer, length, 0);
+        processed_bytes = (bytes_sent == -1) ? 0 : bytes_sent;
+        return (bytes_sent == -1) ? -1 : 0;
+    }
+
+    size_t get_free_space() { return SIZE_MAX; }
+
+private:
+    int _socket_fd;
```

Comment:
the [Google Naming Convention](https://google.github.io/styleguide/cppguide.html#Variable_Names) would be `socket_fd_`

Reply:
maybe add a link saying this to the developer part of the readme.

Reply:
yes, will do after merging this to avoid merge conflicts